python/fledge/services/core/api/service.py

```diff
@@ -183,29 +185,72 @@ async def add_service(request):
                     raise ValueError("format param is required")
                 if file_format not in ["repository"]:
                     raise ValueError("Invalid format. Must be 'repository'")
+                if not name.startswith("fledge-service-"):
+                    raise ValueError('name should start with "fledge-service-" prefix')
                 version = data.get('version', None)
                 if version:
                     delimiter = '.'
                     if str(version).count(delimiter) != 2:
                         raise ValueError('Service semantic version is incorrect; it should be like X.Y.Z')
 
-                services, log_path = await common.fetch_available_packages("service")
-                if name not in services:
-                    raise KeyError('{} service is not available for the given repository or already installed'.format(name))
```

Comment:
We should check this,

Reply:
This piece of code line moves to next position
https://github.com/fledge-iot/fledge/blob/1131dbab6d86e38dc04ed95be958d0adcc643a79/python/fledge/services/core/api/service.py#L227

Reply:
I would suggest to keep it  here itself before going to check status from package management table; and return early in this case.